haystack/document_store/memory.py

```diff
@@ -122,9 +122,9 @@ def update_embeddings(self, retriever: BaseRetriever, index: Optional[str] = Non
         for doc, emb in zip(docs, embeddings):
             self.indexes[index][doc.id].embedding = emb
 
-    def get_document_count(self, index: Optional[str] = None) -> int:
-        index = index or self.index
-        return len(self.indexes[index].items())
+    def get_document_count(self, filters: Optional[Dict[str, List[str]]] = None, index: Optional[str] = None) -> int:
+        documents = self.get_all_documents(index=index, filters=filters)
```

Comment:
What if index=None here? Don't we want to keep `index = index or self.index`?

Reply:
That is taken care by `get_all_documents()`.